### DISCUSSION TITLE
use an http OpenTelemetry exporter instead of gRPC

### DISCUSSION THREAD

**Discussion by jimkont** [quarkusio/quarkus#32155] [Q&A] [answered]
By default, quarkus is using a gRPC-based endpoint to export spans. Is it possible to configure quarkus to use an HTTP endpoint instead?

[a comment by an automated account is not shown]

**Comment by brunobat** [comment, marked as answer]
Hi Dimitris,
We and the OTel community are thinking about making it the default...
For Quarkus 3 we have a new OTel extension and it should be usable on the next CR1 version.
We are not providing oficial support for HTTP, yet but it will contain the implementation you need... If you could help to test drive it it will be great.

**Reply by jimkont** [reply]
Yes, I would be happy to test drive it when it is available

**Reply by brunobat** [reply]
Hi @jimkont CR1 is out, you can try HTTP by setting:
`quarkus.otel.traces.exporter=http`
This was not documented yet because it's still experimental.

**Reply by brunobat** [reply]
Jumped the gun. CR1 is tagged, not out yet.

**Reply by brunobat** [reply]
CR1 is now out and you may test the HTTP exporter.

**Reply by jimkont** [reply]
thanks for the update @brunobat , trying to migrate the code but I am getting some build errors with the quarkus lambda extension, what's the best option for help here? open an issue, start a new discussion, continue here, or some other channel?

**Reply by brunobat** [reply]
If the issue is with the lambda extension you should create a specific discussion or bug, if you think it's not behaving as expected.

**Reply by jimkont** [reply]
Hi @brunobat , quarkus 3.0.3 fixes all the issues I was experiencing and I wanted to try it out now if possible. 
The configuration you mentioned above (`quarkus.otel.traces.exporter=http`) does not work now, is there another way I test the http exporter?

**Reply by brunobat** [reply]
Do you have any error?

**Reply by jimkont** [reply]
I get 
```
 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile [dev]): io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException: Unrecognized value for otel.traces.exporter: http
        at io.opentelemetry.sdk.autoconfigure.SpanExporterConfiguration.configureExporter(SpanExporterConfiguration.java:102)
        at io.opentelemetry.sdk.autoconfigure.SpanExporterConfiguration.configureSpanExporters(SpanExporterConfiguration.java:66)
...
```
with this configuration
```
quarkus:
  otel:
    traces:
      exporter: "http"
      enabled: true
    resource:
      attributes: "service.name=${OTEL_SERVICE_NAME}"
    exporter:
      otlp:
        traces:
          endpoint: ${OTEL_EXPORTER_OTLP_ENDPOINT}
```

**Reply by brunobat** [reply]
Will have to look into it...
